library/ecdsa.c

```diff
@@ -752,8 +752,12 @@ int mbedtls_ecdsa_read_signature_restartable( mbedtls_ecdsa_context *ctx,
 int mbedtls_ecdsa_genkey( mbedtls_ecdsa_context *ctx, mbedtls_ecp_group_id gid,
                   int (*f_rng)(void *, unsigned char *, size_t), void *p_rng )
 {
-    return( mbedtls_ecp_group_load( &ctx->grp, gid ) ||
-            mbedtls_ecp_gen_keypair( &ctx->grp, &ctx->d, &ctx->Q, f_rng, p_rng ) );
+    int ret = 0;
```

Comment:
Oooh, the old code had a bug and the fix should be backported. `f() || g()` returned 1 on failure. This broke application code that looks for errors with `if (mbedtls_ecdsa_genkey(…) < 0)`, which should be correct because the library normally returns negative values on error. This should also be mentioned in the changelog as a bugfix.

Reply:
Backported to 2.7 in https://github.com/ARMmbed/mbedtls/pull/2296